2018 Robot/src/org/usfirst/frc/team4999/robot/subsystems/Lift.java

```diff
@@ -0,0 +1,138 @@
+package org.usfirst.frc.team4999.robot.subsystems;
+
+import org.usfirst.frc.team4999.commands.lift.MaintainLiftHeight;
+import org.usfirst.frc.team4999.robot.RobotMap;
+import org.usfirst.frc.team4999.utils.MoPrefs;
+import org.usfirst.frc.team4999.utils.MomentumPID;
+import org.usfirst.frc.team4999.utils.PIDFactory;
+
+import edu.wpi.first.wpilibj.DoubleSolenoid;
+import edu.wpi.first.wpilibj.Encoder;
+import edu.wpi.first.wpilibj.SpeedControllerGroup;
+import edu.wpi.first.wpilibj.command.Subsystem;
+
+/**
+ *
+ */
+public class Lift extends Subsystem {
+	
+	private static final double MIN_HEIGHT = 0;
+	private static final double MAX_MOTOR_DELTA = 0.05;
+	
+	private SpeedControllerGroup motors = RobotMap.liftMotors;
+	private DoubleSolenoid shifter = RobotMap.liftShifter;
+	private DoubleSolenoid brake = RobotMap.liftBrake;
+	
+	private Encoder encoder = RobotMap.liftEncoder;
+	
+	private MomentumPID slowLiftPID, fastLiftPID;
+	
+	public Lift() {
+		encoder.setDistancePerPulse(1/MoPrefs.getLiftEncTicks());
+		
+		slowLiftPID = PIDFactory.getSlowLiftPID();
+		fastLiftPID = PIDFactory.getFastLiftPID();
+		
+	}
+	
+	public boolean isBraked() {
+		return brake.get() == DoubleSolenoid.Value.kForward;
+	}
+	
+	public void brake() {
+		brake.set(DoubleSolenoid.Value.kForward);
+	}
+	public void releaseBrake() {
+		brake.set(DoubleSolenoid.Value.kReverse);
+	}
+	
+	public boolean isHighSpeed() {
+		return shifter.get() == DoubleSolenoid.Value.kForward;
+	}
+	public void shiftHigh() {
+		shifter.set(DoubleSolenoid.Value.kForward);
+	}
+	public void shiftLow() {
+		shifter.set(DoubleSolenoid.Value.kReverse);
+	}
+	
+	public void driveLiftPID() {
+		if(isHighSpeed())
+			driveFastLiftPID();
+		else
+			driveSlowLiftPID();
+	}
+	
+	public double getCurrentHeight( ) {
+		return encoder.getDistance();
+	}
+	
+	public boolean isOnTarget() {
+		if(isHighSpeed())
+			return fastLiftPID.onTargetForTime();
+		else
+			return slowLiftPID.onTargetForTime();
+	}
+	
+	public void driveSlowLiftPID() {
+		if(slowLiftPID.isEnabled() && !isBraked())
+			set(slowLiftPID.get());
+		else
+			set(0);
+	}
+	public void driveFastLiftPID() {
+		if(fastLiftPID.isEnabled() && !isBraked())
+			set(fastLiftPID.get());
+		else
+			set(0);
+	}
+	public void drivePID() {
+		if(isHighSpeed())
+			driveFastLiftPID();
+		else
+			driveSlowLiftPID();
+	}
+	
+	public void set(double power) {
+		// Anti-jerk: Clip the difference between the requested power and the current power
+		double delta = power - motors.get();
+		delta = clip(delta, -MAX_MOTOR_DELTA, MAX_MOTOR_DELTA);
```

Comment:
How frequently is this method called?  Can you rely on it being called on a regular time basis?  If not, then you should use a reliable time base to damp the power changes.
You also have to be careful with this type of thing because it can fight with the PID and cause unwanted oscillation.